docs/sdk/packages/guides/first-pallet/src/lib.rs

```diff
@@ -0,0 +1,492 @@
+// This file is part of Substrate.
+
+// Copyright (C) Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Pallets use din the `your_first_pallet` guide.
+
+#![cfg_attr(not(feature = "std"), no_std)]
+
+#[docify::export]
+#[frame::pallet(dev_mode)]
+pub mod shell_pallet {
+	// a Shell pallet. This is the bare minimum skeleton of any FRAME-based pallet.
+	use frame::prelude::*;
+
+	/// A trait meant to configure the pallet over types and values. See
+	/// [`frame::pallet_macros::config`] for more info.
+	#[pallet::config]
+	pub trait Config: frame_system::Config {}
+
+	/// The main struct that represents the functionality of the pallet. See
+	/// [`frame::pallet_macros::pallet`] for more info.
+	#[pallet::pallet]
+	pub struct Pallet<T>(_);
+}
+
+#[frame::pallet(dev_mode)]
+pub mod pallet {
+	use frame::prelude::*;
+
+	#[docify::export]
+	pub type Balance = u128;
+
+	#[pallet::config]
+	pub trait Config: frame_system::Config {}
+
+	#[pallet::pallet]
+	pub struct Pallet<T>(_);
+
+	#[docify::export]
+	/// Single storage item, of type `Balance`.
+	#[pallet::storage]
+	pub type TotalIssuance<T: Config> = StorageValue<Value = Balance>;
```

Comment:
I think its good practice to always put an `OptionQuery` or `ValueQuery` explicitly here, so that new devs are aware.  
I often see in SE questions that they dont realize that `OptionQuery` even exists 🙈